src/views/connectionUI.ts

```diff
@@ -195,21 +195,49 @@ export class ConnectionUI {
                 };
             });
 
+            // Add an option to disconnect from the current server
+            const disconnectItem: vscode.QuickPickItem = {
+                label: LocalizedConstants.disconnectOptionLabel,
+                description: LocalizedConstants.disconnectOptionDescription
+            };
+            pickListItems.push(disconnectItem);
+
             const pickListOptions: vscode.QuickPickOptions = {
                 placeHolder: LocalizedConstants.msgChooseDatabasePlaceholder
             };
 
             // show database picklist, and modify the current connection to switch the active database
-            self.vscodeWrapper.showQuickPick<Interfaces.IConnectionCredentialsQuickPickItem>(pickListItems, pickListOptions).then( selection => {
+            self.vscodeWrapper.showQuickPick<vscode.QuickPickItem>(pickListItems, pickListOptions).then( selection => {
+                if (selection === disconnectItem) {
+                    self.handleDisconnectChoice().then(() => resolve(undefined), err => reject(err));
```

Comment:
At this point if you reject what happens? I think it's possible you should handle it by showing the error to the user.